gammapy/spectrum/results.py

```diff
@@ -0,0 +1,351 @@
+        self.flux_errors = flux_errors
+        self.flux_points = Table(flux_points)
+
+    @classmethod
+    def from_fitspectrum_json(cls, filename, model=0):
+        """Read FitSpectrum output file"""
+        import json
+
+        with open(filename) as fh:
+            data = json.load(fh)
+
+        val = data['fit_functions']['fit_functions'][model]
+        e_range = (val['energy_min'], val['energy_max'])
+        energy_range = EnergyBounds(e_range, 'TeV')
+        spectral_model = val['type']
+        parameters = Bunch()
+        parameters_errors = Bunch()
+        for par in val['parameters']:
+            pname = par['name']
+            if pname == 'Index':
+                unit = Unit('')
+                name = 'Gamma'
+            elif pname == 'Norm':
+                unit = val['norm_scale'] * Unit('cm-2 s-1 TeV-1')
+                name = 'Amplitude'
+            elif pname == 'E0':
+                unit = Unit('TeV')
+                name = 'Reference'
+            else:
+                raise ValueError('Unkown Parameter: {}'.format(pname))
+            parameters[name] = par['value'] * unit
+            parameters_errors[name] = par['error'] * unit
+
+        fluxes = Bunch()
+        fluxes['1TeV'] = val['flux_at_1'] * Unit('cm-2 s-1')
+        flux_errors = Bunch()
+        flux_errors['1TeV'] = val['flux_at_1_err'] * Unit('cm-2 s-1')
+        flux_points = Table(data=data['flux_graph']['bin_values'], masked=True)
+        flux_points['energy'].unit = 'TeV'
+        flux_points['energy_err_hi'].unit = 'TeV'
+        flux_points['energy_err_lo'].unit = 'TeV'
+        flux_points['flux'].unit = 'cm-2 s-1 TeV-1'
+        flux_points['flux_err_hi'].unit = 'cm-2 s-1 TeV-1'
+        flux_points['flux_err_lo'].unit = 'cm-2 s-1 TeV-1'
+        flux_points['flux_ul'].unit = 'cm-2 s-1 TeV-1'
+
+        return cls(energy_range=energy_range, parameters=parameters,
+                   parameter_errors=parameters_errors,
+                   spectral_model=spectral_model,
+                   fluxes=fluxes, flux_errors=flux_errors,
+                   flux_points=flux_points)
+
+    @classmethod
+    def from_sherpa(cls, covar, filter, model):
+        """Create SpectrumFitResult from sherpa objects"""
+        el, eh = float(filter.split(':')[0]), float(filter.split(':')[1])
+        energy_range = EnergyBounds((el, eh), 'keV')
+        if model.name.split('.')[0] == 'powlaw1d':
+            spectral_model = 'PowerLaw'
+        else:
+            raise ValueError("Model not understood: {}".format(model.name))
+        parameters = Bunch()
+        parameter_errors = Bunch()
+
+        # Todo: Support assymetric errors
+        for pname, pval, perr in zip(covar.parnames, covar.parvals,
+                                     covar.parmaxes):
+            pname = pname.split('.')[-1]
+            if pname == 'gamma':
+                name = 'Gamma'
+                unit = Unit('')
+            elif pname == 'ampl':
```

Comment:
Didn't we discuss about assigning parameter names for serialized results based on 
gamma-astro-data-formats/source/source-models/spectral_models.yaml 
?